hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -212,7 +211,9 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
     } else {
       val (recordName, namespace) = AvroConversionUtils.getAvroRecordNameAndNamespace(table.identifier.table)
       val schema = SchemaConverters.toAvroType(dataSchema, nullable = false, recordName, namespace)
-      val partitionColumns = if (table.partitionColumnNames.isEmpty) {
+      val partitionColumns = if (tableConfigs.contains(KeyGeneratorOptions.PARTITIONPATH_FIELD_NAME.key())) {
```

Comment:
Why this instead of HoodieTableConfig.PARTITION_FIELDS?

Reply:
This is while creation of hudi table. So I have used the config used for setting partition fields. 
I dont think we suggest using HoodieTableConfig.PARTITION_FIELDS while creation of new table?

Reply:
thanks, that makes sense.